### PR TITLE
chore: 1.19.3 port

### PR DESCRIPTION
**Pull request by kristofbolyai**
This PR ports the whole mod to 1.19.3.

### CODE REVIEW

**Comment by magicus** [comment]
Okay, so the time has finally come to take the bull by the horns... upgrading Minecraft version.

Are you thinking we should have a 1.19 and an 1.18 version, or should we drop 1.18?

**Comment by magicus** [comment]
By a quick glance it looked like most changes were due to Component.literal. If we had our own wrapper, which could be compiled differently on 1.18 and 1.19, I wonder how many changes there would be then. Maybe few enough that it could be worth supporting several versions? 

I don't know what kind of interest there is in keeping an 1.18 version of Artemis, but I'm guessing the controversial "chat reporting" system in 1.19 is making at least some users queasy about using 1.19 on servers.

**Comment by magicus** [comment]
(And finally, regarding the PR tags -- this is a major undertaking and functionality change for the end users, and does not seems fittingly described by "chore")

**Comment by kristofbolyai** [comment]
> By a quick glance it looked like most changes were due to Component.literal. If we had our own wrapper, which could be compiled differently on 1.18 and 1.19, I wonder how many changes there would be then. Maybe few enough that it could be worth supporting several versions? 
> 
> 
> 
> I don't know what kind of interest there is in keeping an 1.18 version of Artemis, but I'm guessing the controversial "chat reporting" system in 1.19 is making at least some users queasy about using 1.19 on servers.

Supporting both versions is much harder than you would expect on first look. Based on the community vote I organised, 1.19 is prefered by far.

**Comment by magicus** [comment]
Did you try using the fabric migration tool? `gradlew migrateMappings --mappings "1.19.2+build.28"` It's in no way perfect, but my impression is that it does a decent job of automatically converting some refactorings Mojang has done.

**Comment by kristofbolyai** [comment]
> Did you try using the fabric migration tool? `gradlew migrateMappings --mappings "1.19.2+build.28"` It's in no way perfect, but my impression is that it does a decent job of automatically converting some refactorings Mojang has done.

I did not, but I did not need to. CTRL + F solved almost every problem :). The things it wouldn't solve would require my manual attention either way.

**Comment by magicus** [comment]
For future upgrades it can be worth remembering. It makes a copy in "updatedSrc" (or something like that), and then you can compare it for any changes it makes, and chose to keep them if you think they are good.

File: fabric/build.gradle [hunk 1]
```diff
@@ -18,7 +18,7 @@ buildscript {
          */
 
         dependencies {
-            classpath "com.github.kristofbolyai:EventBusTransformer:3fedef0169"
```
Comment (magicus) [inline fabric/build.gradle]: It's just weird that a central dependency -- that we develop ourselves -- should reside in personal name space.
I forked your repo into https://github.com/Wynntils/EventBusTransformer, so you can switch to that instead.

**Comment by kristofbolyai** [comment]
@magicus I see you created an official fork of the event bus transformer. My version is a bit patchy, but coehlrich is working on a bit cleaner version.

**Comment by kristofbolyai** [comment]
> @magicus I see you created an official fork of the event bus transformer. My version is a bit patchy, but coehlrich is working on a bit cleaner version.

I've let them known to use the official repo now. Anyways, a centralized repo is always better.

**Comment by magicus** [comment]
I picked your repo as starting point since you updated this patch to use this version instead. I thought @coehlrich did not have time to work on it anymore. Do you want me to re-create the official repo starting with coehlrich's version instead?

**Comment by kristofbolyai** [comment]
> I picked your repo as starting point since you updated this patch to use this version instead. I thought @coehlrich did not have time to work on it anymore. Do you want me to re-create the official repo starting with coehlrich's version instead?

It is fine I think, their repo is in the fork tree.

**Review by magicus — CHANGES_REQUESTED** [review]
Please re-submit, targeting `main`. Sorry for the trouble. 😢

**Comment by kristofbolyai** [comment]
Today marks the point where we have a building and loading 1.19.3 version, without obvious crashes.

File: common/src/main/java/com/wynntils/mc/EventFactory.java [hunk 2]
```diff
@@ -497,27 +506,34 @@ public static SetSpawnEvent onSetSpawn(BlockPos spawnPos) {
...
+            for (ClientboundPlayerInfoUpdatePacket.Action action : packet.actions()) {
+                if (action == ClientboundPlayerInfoUpdatePacket.Action.UPDATE_DISPLAY_NAME) {
+                    GameProfile profile = entry.profile();
+                    if (entry.displayName() == null) continue;
+                    post(new PlayerDisplayNameChangeEvent(profile.getId(), entry.displayName()));
+                }
             }
         }
+
+        //            else if (action == Action.REMOVE_PLAYER) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/mc/EventFactory.java]: @kristofbolyai  This looks like a "FIXME". I assume we still want to post PlayerLogOutEvents...

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/mc/EventFactory.java]: No, I just forgot to remove this. If you look at this change, logout and other actions were separated with packets. Just below this is where I handle this.

**Pull request closed without merging**